cg/services/illumina/file_parsing/models.py

```diff
@@ -28,3 +29,41 @@ class DemuxMetrics(BaseModel):
     lane: int = Field(..., alias=DemuxMetricsColumnNames.LANE)
     sample_internal_id: str = Field(..., alias=DemuxMetricsColumnNames.SAMPLE_INTERNAL_ID)
     read_pair_count: int = Field(..., alias=DemuxMetricsColumnNames.READ_PAIR_COUNT)
+
+
+class DsmcEncryptionKey(BaseModel):
```

Comment:
There are a couple of things I would change about the Pydantic classes in this file, although nothing pertains to the logic and your classes should probably be able to do what you want. 

1. Make sure that the type hints for the fields are the types you intend to _use them as_, e.g. `date` should be type hinted as `datetime`. You might run into parsing errors since the input seems to be a string but we shall fix that at a later stage. The type hint should always refer to what a user of the class can expect the type to be.
2. In order to make step 1 work, we need to convert the input strings to `datetime` objects (and potentially `Path` objects if that is how we want them). This can be done in the validators you have added, by setting `mode="before"` although I would like the validators refactored a bit as well.
3. We are trying to write Pydantic BaseModels in a unified and clear way. Part of that has been to agree to using another way of writing validators, similar to how it is described here https://docs.pydantic.dev/latest/concepts/validators/#annotated-validators. In essence what we want to do is to keep the BaseModels as clear of logic as possible but still understandable in what logic is being performed. Thus we would not want the methods for the validators written in the classes and we would not want to read all the different validator methods to see which one is actually applied to which field.

You can solve this by writing your `parse_date` method outside of the class, either in this file or ideally in a validators.py file in the same directory. It might be possible that there are previous implementations of the methods, e.g. `parse_date` in which case they can be reused.

Given that you have declared your methods that you will use for validation, you can specify in the type hint both what data type the field is and what validations are being performed by using `Annotated` coupled with the `Validator` classes provided by Pydantic. 

I will give suggestions on the actual fields to give the syntax.